internal/core/queries/device_compatibility.go

```diff
@@ -39,6 +47,33 @@ func NewGetDeviceCompatibilityQueryHandler(dbs func() *db.ReaderWriter, reposito
 	}
 }
 
+func GetDeviceCompatibilityLevel(fd map[string]FeatureDetails, totalWeights float64) string {
+	level := ""
+	total := 0.0
+
+	for _, v := range fd {
+		if v.SupportLevel > 0 {
+			total += v.FeatureWeight
+		} else {
+			total += 0
+		}
+	}
+
+	if total != 0 && total < totalWeights {
```

Comment:
question: Shouldn't `total = totalWeights` be gold?

suggestion: It might be good to warn consumers in the `.proto` that the `level` field might be an empty string.

I agree that "bronze" does not quite describe "has no useful signals whatsoever".

Reply:
right bronze means there is some degree of compatibility but not a whole lot, eg. cars before 2008 with AutoPi